core/src/main/java/jenkins/util/VirtualFile.java

```diff
@@ -62,6 +75,26 @@
  * {@link VirtualFile} makes no assumption about where the actual files are, or whether there really exists
  * {@link File}s somewhere. This makes VirtualFile more abstract.
  *
+ * <h2>Opening files from other machines</h2>
+ *
+ * While {@link VirtualFile} is marked {@link Serializable},
+ * it is <em>not</em> safe in general to transfer over a Remoting channel.
+ * (For example, an implementation from {@link #forFilePath} could be sent on the <em>same</em> channel,
+ * but an implementation from {@link #forFile} will not.)
+ * Thus callers should assume that methods such as {@link #open} will work
+ * only on the node on which the object was created.
+ *
+ * <p>Since some implementations may in fact use external file storage,
+ * callers may request optional APIs to access those services more efficiently.
```

Comment:
🐜 for ignoring `InterruptedException` in API. In FS-based implementation they are now wrapped by IOException, which may lead to improper suppression by the logic (aka "user fails to interrupt the thread and scratches his head").

The same applies for other implementations I may imagine, because the storages are slow. So I would rather expect all APIs to support `InterruptedException`

Reply:
I still feel that we grow a technical debt in API, but Ican live with that.